03.api.md

```diff
@@ -358,92 +358,973 @@ jerry_api_gc ();
+
+  ... // usage of string
+
+  jerry_api_release_string(string);
+}
+
+```
+
+**See also**
+
+- [jerry_api_release_string](#jerry_api_release_string)
+
+
+# jerry_api_create_string_value
+
+**Summary**
+
+Creates a JERRY_API_DATA_TYPE_STRING type jerry_api_value_t from the given jerry_api_string_t parameter. The value have to be released by [jerry_api_release_value](#jerry_api_release_value) when it becomes unnecessary.
+
+**Prototype**
+
+```c
+jerry_api_value_t
+jerry_api_create_string_value (jerry_api_string_t *value);
+```
+- `value` - jerry_api_string_t from which a value will be created
+- return value - a jerry_api_value_t created from the given jerry_api_string_t argument
+
+**Example**
+
+```c
+{
+  jerry_init (JERRY_FLAG_EMPTY);
+  jerry_api_char_t char_array[] = "a string";
+  jerry_api_string_t *string  = jerry_api_create_string_sz (char_array,
+                                                            lit_zt_utf8_string_size (char_array));
+  jerry_api_value_t string_value = jerry_api_create_string_value(string);
+
+  ... // usage of string_value
+
+  jerry_api_release_value(&string_value);
+}
+```
+
+**See also**
+
+- [jerry_api_value_t](#jerry_api_value_t)
+- [jerry_api_release_value](#jerry_api_release_value)
+
+
+# jerry_api_create_error_sz
+
+**Summary**
+
+Creates an error object. Caller should release the object with jerry_api_release_object, just when the value becomes unnecessary.
+
+**Prototype**
+
+```c
+jerry_api_object_t *
+jerry_api_create_error_sz (jerry_api_error_t error_type,
+                           const jerry_api_char_t *message_p,
+                           jerry_api_size_t message_size);
+```
+- `error_type` - type of the error
+- `message_p` - value of 'message' property of the constructed error object
+- `message_size` - size of the message in bytes
+- return value - pointer to created error object
+
+**Example**
+
+```c
+{
+  jerry_init (JERRY_FLAG_EMPTY);
+  jerry_api_char_t message[] = "error";
+  jerry_api_object_t *error_obj = jerry_api_create_error_sz(JERRY_API_ERROR_COMMON,
+                                                            message,
+                                                            lit_zt_utf8_string_size (char_array));
+
+  ... // usage of error_obj
+
+  jerry_api_release_object(error_obj);
+}
+```
+
+**See also**
+
+- [jerry_api_create_error](#jerry_api_create_error)
+
+
+# jerry_api_get_boolean_value
+
+**Summary**
+
+Gets the raw bool value form a jerry_api_valu_t value.
+If the given jerry_api_value_t structure has type other than JERRY_API_DATA_TYPE_BOOLEAN, JERRY_ASSERT fails.
+
+**Prototype**
+
+```c
+bool
+jerry_api_get_boolean_value (const jerry_api_value_t *value_p);
+```
+- `value_p` - pointer to an api value
+- return value - the boolean v_bool member of the given jerry_api_value_t structure
+
+**Example**
+
+```c
+{
+  jerry_init (JERRY_FLAG_EMPTY);
+  jerry_api_value_t value;
+  ... // create or acquire value
+
+  if (jerry_api_value_is_boolean (&value))
```

Comment:
Yes, we have these functions. I think we should not expose the union then. I mean we can expose it if necessary, but not document it, because it is internal, and can be changed anytime.